python/lsst/ts/ess_sensors/sel_temperature_reader.py

```diff
@@ -0,0 +1,270 @@
+# This file is part of ts_ess_sensors.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+"""Implementation of SEL temperature instrument.
+
+Read and perform protocol conversion for Straight Engineering Limited (SEL)
+temperature measurement instrument.
+SEL temperature instruments output serial data in lines terminated by '\r\n'.
+There are multiple SEL temperature instruments with different numbers of
+channels. The number of channels must be specified at instantiation.
+The format of a 4-channel instruments output line is:
+
+    'C00=snnn.nnnn,C01=snnn.nnnn,C02=snnn.nnnn,C03=snnn.nnnn\r\n'
+
+where:
+
+    'C00='      4-character preamble with Celcius ('C') and
+                two digit channel number.
+
+    'snnn.nnnn' 9-character decimal value with negative sign or positive value
+                (s = '-' or '0..9') and decimal point in the fifth position.
+
+    ','         Single character delimiter.
+
+    '\r\n'      2-character terminator.
+"""
+
+__all__ = ["SelTemperature", "DELIMITER"]
+
+import asyncio
+import logging
+import time
+from typing import Any, Dict
+
+from .serial_reader import SerialReader
+
+# Instrument serial parameters ....
+BAUDRATE: int = 19200
```

Comment:
Please consider SEL_BAUDRATE or similar, so this can be a public constant. (Pity SEL and SAL look so similar, but at this this is a SAL-free package).